pandas/core/indexes/interval.py

```diff
@@ -440,23 +440,31 @@ def is_monotonic(self):
         Return True if the IntervalIndex is monotonic increasing (only equal or
         increasing values), else False
         """
-        return self._multiindex.is_monotonic
+        return self.is_monotonic_increasing
 
     @cache_readonly
     def is_monotonic_increasing(self):
         """
         Return True if the IntervalIndex is monotonic increasing (only equal or
         increasing values), else False
         """
-        return self._multiindex.is_monotonic_increasing
+        values = [self.right, self.left]
+
+        try:
+            sort_order = np.lexsort(values)
+            return Index(sort_order).is_monotonic
+        except TypeError:
+
+            # we have mixed types and np.lexsort is not happy
```

Comment:
Can this situation actually occur for `IntervalIndex`?  If I'm interpreting this correctly, we shouldn't have mixed types in an `IntervalIndex` as a consistent type is enforced during construction.

Reply:
are there tests which hit this situation?

Reply:
`IntervalIndex._data` is already an `IntervalArray` in which the check on the same types is present.

```
In [2]: left = [1, 2, 3]                                                        

In [3]: right = pd.date_range('2019-01-01', periods=3, freq='D')                

In [4]: intv = pd.IntervalIndex.from_arrays(left, right)                        
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-4-50e34a81b4a8> in <module>
----> 1 intv = pd.IntervalIndex.from_arrays(left, right)

~/Code/pandas/pandas/core/indexes/interval.py in from_arrays(cls, left, right, closed, name, copy, dtype)
    187         with rewrite_exception("IntervalArray", cls.__name__):
    188             array = IntervalArray.from_arrays(left, right, closed, copy=copy,
--> 189                                               dtype=dtype)
    190         return cls._simple_new(array, name=name)
    191 

~/Code/pandas/pandas/core/arrays/interval.py in from_arrays(cls, left, right, closed, copy, dtype)
    325 
    326         return cls._simple_new(left, right, closed, copy=copy,
--> 327                                dtype=dtype, verify_integrity=True)
    328 
    329     _interval_shared_docs['from_intervals'] = """

~/Code/pandas/pandas/core/arrays/interval.py in _simple_new(cls, left, right, closed, copy, dtype, verify_integrity)
    190                    '[{rtype}] types')
    191             raise ValueError(msg.format(ltype=type(left).__name__,
--> 192                                         rtype=type(right).__name__))
    193         elif is_categorical_dtype(left.dtype) or is_string_dtype(left.dtype):
    194             # GH 19016

ValueError: must not have differing left [Int64Index] and right [DatetimeIndex] types
```
There are not tests for mixed types in IntervalIndex and IntervalArray. Should we add in a new PR?

Reply:
Tests for the constructors can go in a separate PR